src/services/subscription.ts

```diff
@@ -0,0 +1,54 @@
+import { Service } from 'typedi'
+
+@Service()
+export default class SubscriptionService {
+    private readonly subscriptions: Map<string, Array<string>>
+
+    constructor () {
+      this.subscriptions = new Map()
```

Comment:
Can you move this map to `redis`? That way the storage will be both efficient and persistant